samples: modem_shell: Disable modem traces before disabling UART1 (#7448)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

        This includes several new fields in the PVT notification and a command to query the expiry times of assistance data.
      * Support for the :kconfig:option:`CONFIG_NRF_CLOUD_PGPS_STORAGE_PARTITION` option.

    * Updated:

>       * When :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_UART` is enabled, modem traces will be disabled before disabling UART1. They will be re-enabled when re-enabling UART1.

Reviewer: ```suggestion
      * When the :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_UART` option is enabled, modem traces are also disabled if UART1 is disabled.
        When you enable UART1 again, the modem traces are re-enabled.
```
Not at all sure I managed to keep the idea right. Anyway, the original seems very strange.

Reviewer 2: I think the idea is mostly preserved. The UART1 can either be re-enabled on a timer expiry or by the user pressing a button. So the 'you' in second sentence may not be correct. Can I write the following?

`When the UART1 is re-enabled (either after timer expiry or button press), the modem traces are also re-enabled.`

Reviewer 2: Thanks 👍  will fix

        This includes several new fields in the PVT notification and a command to query the expiry times of assistance data.
      * Support for the :kconfig:option:`CONFIG_NRF_CLOUD_PGPS_STORAGE_PARTITION` option.

    * Updated:

      * The behavior of this sample when built with the :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_UART` option enabled, is changed as follows:
        When disabling of UART is requested either by a shell command or by a button press, modem traces are disabled before disabling UART1.

>         When the UART1 is re-enabled (either after timer expiry or button press), the modem traces are also re-enabled.

Reviewer: Have these two as bullet points, too. Indented two two empty spaces. And leave a blank line after the first line ending with colon.
